src/c++/perf_analyzer/triton_metrics_manager.cc

```diff
@@ -81,13 +85,22 @@ TritonMetricsManager::QueryTritonMetricsEveryNMilliseconds()
   }
 }
 
+void
+TritonMetricsManager::CheckQueryingStatus()
+{
+  if (query_loop_future_.wait_for(std::chrono::seconds(0)) ==
+      std::future_status::ready) {
+    query_loop_future_.get();
+  }
+}
+
 void
 TritonMetricsManager::StopQueryingTritonMetrics()
 {
   should_keep_querying_ = false;
   query_loop_cv_.notify_one();
   if (query_loop_future_.valid()) {
-    query_loop_future_.wait();
+    query_loop_future_.get();
```

Comment:
`std::future::wait()` doesn't propagate exceptions, but `std::future::get()` does.

We still want to propagate any potential exceptions thrown during the future's asynchronous execution if an exception happened between the last time `CheckQueryingStatus()` was called and the end of profiling in general (when `StopQueryingTritonMetrics()` gets called).